yb-voyager/cmd/root.go

```diff
@@ -120,3 +135,36 @@ func validateExportDirFlag() {
 		exportDir = strings.TrimRight(exportDir, "/")
 	}
 }
+
+func lockExportDir() {
+	lockfileName, err := filepath.Abs(filepath.Join(exportDir, ".lockfile.lck"))
+	if err != nil {
+		fmt.Printf("Failed to get the lockfile path: %v", err)
+		os.Exit(1)
+	}
+
+	lock, err = lockfile.New(lockfileName)
+	if err != nil {
+		fmt.Printf("Failed to create lockfile %q: %v", lockfileName, err)
+		os.Exit(1)
+	}
+
+	err = lock.TryLock()
+	if err == nil {
+		return
+	} else if err == lockfile.ErrBusy {
+		fmt.Println("Another instance of yb-voyager is running in the export-dir.")
+		os.Exit(1)
+	} else {
+		fmt.Printf("Unable to lock the export-dir: %v", err)
+		os.Exit(1)
+	}
+
+}
+
+func unlockExportDir() {
+	err := lock.Unlock()
+	if err != nil {
+		panic(fmt.Sprintf("Unable to unlock %q: %v", lock, err))
```

Comment:
can we use fmt.Printf() here also instead of panic

Reply:
Yeah I'll add printf and os.Exit() instead of panic.